### PR TITLE
swagger codegen added

### PR DESCRIPTION
**Pull request by keep-it-unreal**

### CODE REVIEW

File: src/codegen/swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+        400:
+          description: "Invalid roles name supplied"
+        404:
+          description: "Roles name not found"
+        405:
+          description: "Validation exception"
+  /permission/findByName/{permissionName}:
+    get:
+      tags:
+        - "Permission"
```
Comment (tayareut) [inline src/codegen/swagger.yaml:248]: tag name needs to start with a lowercase letter

File: src/codegen/swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            type: "string"
+            items:
+              $ref: "#/definitions/Permission"
+        400:
+          description: "Invalid roles name value"
+  /user/deleteByPermissionsName/{PermissionsName}:
```
Comment (tayareut) [inline src/codegen/swagger.yaml:268]: PermissionsName needs to start with a lowercase letter

File: src/codegen/swagger.yaml
Comment (tayareut) [inline src/codegen/swagger.yaml:1]: Why roles-permissions and users-roles tables are not described? They will be separate entities as well
We use UUID in our approved database schema for all entities
For Roles and Permissions we have is_active field in db schema. If we don't need it, we can agree and I'll remove it from my tasks. But it is already in the schema and I’ve taken it into account when creating the database: https://medicalcenterv1.atlassian.net/wiki/spaces/NP/pages/1605738/Role+model
Can we take into consideration all our restrictions - Not Null, Unique, min and max length, default? Or is it optional?
The get all endpoint is missing for all objects

File: src/codegen/swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,371 @@
+---
+swagger: "2.0"
```
Comment (tayareut) [inline src/codegen/swagger.yaml:2]: Why did you decide to use version 2 instead of the latest one?

File: src/codegen/swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+  description: "This is a sample server role-service."
+  version: "1.0.0"
+  title: "Swagger role-service"
+  termsOfService: "http://swagger.io/terms/"
+  contact:
+    email: "wit2022.v1@gmail.com>"
+  license:
+    name: "Apache 2.0"
+    url: "http://www.apache.org/licenses/LICENSE-2.0.html"
+host: "medicalcenter.roleservice"
```
Comment (tayareut) [inline src/codegen/swagger.yaml:13]: I think we need to have the actual hostname that we set using docker compose.

File: src/codegen/swagger.yaml [hunk 5]
```diff
@@ -0,0 +1,371 @@
...
+basePath: "/v1"
+tags:
+  - name: "User"
```
Comment (tayareut) [inline src/codegen/swagger.yaml:16]: The tag names should start with a lowercase letter - "User" should be "user".

File: src/codegen/swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,371 @@
...
+    description: "Users"
+  - name: "role"
+    description: "Users roles"
```
Comment (tayareut) [inline src/codegen/swagger.yaml:19]: Perhaps a more detailed description of roles and permissions might be useful?

File: src/codegen/swagger.yaml [hunk 7]
```diff
@@ -0,0 +1,371 @@
...
+        - "application/json"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "body"
+          name: "body"
+          description: "Role object that needs to be added to the service"
+          required: true
+          schema:
+            $ref: "#/definitions/User"
```
Comment (tayareut) [inline src/codegen/swagger.yaml:150]: ref is incorrect

File: src/codegen/swagger.yaml [hunk 8]
```diff
@@ -0,0 +1,371 @@
...
+      responses:
+        400:
+          description: "Invalid ID supplied"
+        404:
+          description: "User not found"
+        405:
+          description: "Validation exception"
+  /user/findByExternalID/{externalID}:
```
Comment (tayareut) [inline src/codegen/swagger.yaml:71]: get by id, not externalId

File: src/codegen/swagger.yaml [hunk 9]
```diff
@@ -0,0 +1,371 @@
...
+          schema:
+            $ref: "#/definitions/User"
+      responses:
+        400:
+          description: "Invalid roles name supplied"
+        404:
+          description: "Roles name not found"
+        405:
+          description: "Validation exception"
+  /role/findByName/{rolesName}:
```
Comment (tayareut) [inline src/codegen/swagger.yaml:158]: get by id

File: src/codegen/swagger.yaml [hunk 10]
```diff
@@ -0,0 +1,371 @@
...
+          schema:
+            $ref: "#/definitions/Permission"
+      responses:
+        400:
+          description: "Invalid roles name supplied"
+        404:
+          description: "Roles name not found"
+        405:
+          description: "Validation exception"
+  /permission/findByName/{permissionName}:
```
Comment (tayareut) [inline src/codegen/swagger.yaml:245]: get by id

File: src/codegen/swagger.yaml [hunk 11]
```diff
@@ -0,0 +1,371 @@
...
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            type: "string"
+            items:
+              $ref: "#/definitions/User"
+        400:
+          description: "Invalid externalID value"
+  /user/deleteByExternalID/{externalID}:
```
Comment (tayareut) [inline src/codegen/swagger.yaml:94]: delete by id

File: src/codegen/swagger.yaml [hunk 12]
```diff
@@ -0,0 +1,371 @@
...
+      responses:
+        200:
+          description: "successful operation"
+          schema:
+            type: "string"
+            items:
+              $ref: "#/definitions/Role"
+        400:
+          description: "Invalid roles name value"
+  /role/deleteByRolesName/{rolesName}:
```
Comment (tayareut) [inline src/codegen/swagger.yaml:181]: delete by id

File: src/codegen/swagger.yaml (hunk 2 above)
Comment (tayareut) [reply src/codegen/swagger.yaml:268]: delete by id

File: src/codegen/swagger.yaml
Comment (keep-it-unreal) [reply src/codegen/swagger.yaml:1]: I think that roles-permissions and users-roles entities don't exist. These relations will be created by spring. isActive added) If you have idea how to add "not null" and other restrictions, I ready to make changes))

File: src/codegen/swagger.yaml (hunk 4 above)
Comment (keep-it-unreal) [reply src/codegen/swagger.yaml:13]: What do you mean?

File: src/codegen/swagger.yaml
Comment (tayareut) [reply src/codegen/swagger.yaml:1]: hope this helps with restrictions) 
https://swagger.io/docs/specification/describing-parameters/

**Pull request closed without merging**